language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java

```diff
@@ -71,11 +71,12 @@
 public class CreateVariableWithTypeCodeAction extends CreateVariableCodeAction {
 
     private static final String NAME = "Create variable with type";
+    private static final String DIAGNOSTIC_MESSAGE_KEY = "cannot.infer.typedesc.argument.without.cet";
 
     @Override
     public boolean validate(Diagnostic diagnostic, DiagBasedPositionDetails positionDetails,
                             CodeActionContext context) {
-        return diagnostic.diagnosticInfo().code().equals("BCE4046")
+        return diagnostic.diagnosticInfo().messageFormat().equals(DIAGNOSTIC_MESSAGE_KEY)
```

Comment:
Is using the `message key` more reliable than using diagnostic codes? Ideally diagnostic codes aren't supposed to change frequently. If a diagnostic code changes (or new diagnostic codes are added), adding/modifying the diagnostic code here is the best approach IMO.

Reply:
Yes, +1. We are gradually moving away from diagnostic messages and using codes in code actions. Hence, let's change the fix accordingly please.

Reply:
In this scenario, due to the revert of client declaration related changes diagnostic codes are changed.

Reply:
Yes, other thing is, there can be 2 diagnostics with the same message, but with 2 different diagnostic codes and different properties. We rely on diag properties to determine some contextual information.

Reply:
Update the changes.